### PR TITLE
Dismantle CGI module

### LINKED ISSUE #1511: RFC: Move functions from `CGI` module to their respective places in stdlib
Extracting these ideas from #1505 

```crystal
module HTTP::Form
  # parses www-form-urlencoded string, former CGI.parse
  def self.parse(query : String) : Hash(String, Array(String))
  def self.parse(query : String, &block : (String, String) ->)

  # builds www-form-urlencoded string, former CGI.build_form
  def self.build(&block : FormBuilder ->) : String
end

class URI
  # escapes provided string, former CGI.escape
  def self.escape(string : String) : String
  def self.escape(string : String, io : IO)

  # unescapes provided string, former CGI.unescape
  def self.unescape(string : String) : String
  def self.unescape(string : String, io : IO)
end
```

WDYT?

/cc @asterite @jhass

### PR DESCRIPTION
**Pull request by waterlink**
fixes #1511 

- move `escape`, `unescape` to `URI`
- move `parse`, `build_form` (as `build`) to newly created `HTTP::Params`

/cc @asterite @jhass

### CODE REVIEW

**Comment by ysbaddaden** [comment]
What about HTTP::Params? I would never think to search for HTTP::Form.

**Comment by asterite** [comment]
I like this! :+1: 

About the name, Form could work, at least it's mentioned [here](https://en.wikipedia.org/wiki/Query_string#Web_forms) and it's known from "posting form data". I think they are also query params... I'm not sure what's the "official" name for this, though.

File: spec/std/uri_spec.cr [hunk 1]
```diff
@@ -61,5 +61,58 @@ describe "URI" do
...
+      {"hello%20world", "hello world"},
+      {"hello%", "hello%"},
+      {"hello%2", "hello%2"},
+      {"hello%2B", "hello+"},
+      {"hello%2Bworld", "hello+world"},
+      {"hello%2%2Bworld", "hello%2+world"},
+      {"%E3%81%AA%E3%81%AA", "なな"},
+      {"%e3%81%aa%e3%81%aa", "なな"},
+      {"%27Stop%21%27+said+Fred", "'Stop!' said Fred"},
+    ].each do |tuple|
```
Comment (jhass) [inline spec/std/uri_spec.cr]: Super minor, but could make the outer one  a tuple too ;P

**Comment by jhass** [comment]
On second thought I like `HTTP::Params` but I wouldn't mind `HTTP::Form` either, so LGTM :+1:

**Comment by waterlink** [comment]
Hmm, I kinda like `HTTP::Params` too. Will change it to it

**Comment by waterlink** [comment]
Updated according to suggestions

**Comment by jhass** [comment]
Thank you!

**Comment by asterite** [comment]
With waj we liked forn because it's closer to the original name and it's shorter, but maybe params is more known and familiar, so it's ok :)

**Comment by Nakilon** [comment]
We don't like making aliases?

**Comment by waterlink** [comment]
We don't: Aliases means more things to learn and more confusion when jumping between codebases, or even when the same thing under different names is used in the same codebase.

**Pull request merged**